.azure-pipelines-templates/deploy_aci.yml

```diff
@@ -0,0 +1,44 @@
+jobs:
+  - job: deploy_aci
+    displayName: "Deploy ACI"
+    pool:
+      vmImage: ubuntu-20.04
+    steps:
+      - script: |
+          set -ex
+          ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa -N ""
+          echo "##vso[task.setvariable variable=hostPubKey]`cat ~/.ssh/id_rsa.pub`"
+          echo "##vso[task.setvariable variable=hostPrivKey;isOutput=true;issecret=true]`base64 -w 0 ~/.ssh/id_rsa`"
+        name: generate_ssh_key
+
+      - template: azure_cli.yml
+
+      - script: |
+          set -ex
+          python3.8 scripts/azure_deployment/arm_template.py deploy aci --subscription-id $(CCF_AZURE_SUBSCRIPTION_ID) --resource-group ccf-aci --aci-type dynamic-agent --deployment-name ci-$(Build.BuildNumber) --aci-ssh-keys "$(generate_ssh_key.hostPubKey)" > ~/aci_ips
+          echo "##vso[task.setvariable variable=ipAddresses;isOutput=true]`cat ~/aci_ips`"
+        name: deploy_aci
+        env:
+          CCF_AZURE_SUBSCRIPTION_ID: $(CCF_AZURE_SUBSCRIPTION_ID)
+
+      - script: |
+          set -ex
+          IFS='\n' read -ra IP_ADDR_LIST <<< "$(deploy_aci.ipAddresses)"
+          for IP_ADDR in "${IP_ADDR_LIST[@]}"; do
+            echo -e "Testing connection with $IP_ADDR"
+            for i in {1..3}; do ssh agent@$IP_ADDR -o "StrictHostKeyChecking no" 'echo "Connected successfully"' && break || echo "Connection failed, ACI may still be spinning up, retrying"; done
```

Comment:
I think a comment here to explain the loop would be useful.

Reply:
The retry loop? Does the error message cover that?